contracts/CitizenERC721.sol

```diff
@@ -17,13 +18,14 @@ contract CitizenERC721 is Initializable, ERC721Upgradeable, ERC721EnumerableUpgr
     bytes32 public constant UPGRADER_ROLE = keccak256("UPGRADER_ROLE");
     bytes32 public constant DEVICE_ROLE = keccak256("DEVICE_ROLE");
     CountersUpgradeable.Counter private _tokenIdCounter;
+    CitizenENSRegistrar public _ensRegistrar;
 
     // Allow the baseURI to be updated.
     string private _baseUpdateableURI;
 
     event UpdateBaseURI(string baseURI);
 
-    function initialize() initializer public {
+    function initialize(CitizenENSRegistrar ensRegistrar) initializer public {
```

Comment:
Ok, after review, this looks right to me in order to ensure we can upgrade the registrar in the future (is that the goal of passing via initialize?)

Reply:
To be honest, I just added that there because used to initializing variables through the constructor.

Happy to hardcode it as well.

Reply:
Okay @ccamrobertson, so after testing this, we can't actually edit the `initialize` function (makes sense lol).

So I've gone ahead and added a `setRegistrarAddress` function that updates the variable 😄